backends/qualcomm/quantizer/utils.py

```diff
@@ -119,15 +119,19 @@ def annotate_binary(node: Node, quantization_config: QuantizationConfig) -> None
         return
 
     input_act_qspec = quantization_config.input_activation
-    output_act_qspec = quantization_config.output_activation
+    output_act_qspec = (
+        quantization_config.output_activation
+        if node.meta["val"].dtype == torch.float32
+        else None
+    )
 
     input_qspec_map = {}
     input_act0 = node.args[0]
-    if isinstance(input_act0, Node):
+    if isinstance(input_act0, Node) and input_act0.meta["val"].dtype == torch.float32:
```

Comment:
We can have a seperate PR for this line too. https://github.com/pytorch/executorch/pull/2957 has a few more checks that we may need.

Reply:
Yes, this looks like a common issue.